lib/features/pokemon_overview/pokemon_overview_page.dart

```diff
@@ -1,45 +1,109 @@
+  late TextEditingController _searchController;
+  late bool isSearching;
+  Timer? _debounceTimer;
+
+  @override
+  void initState() {
+    super.initState();
+
+    _searchController = TextEditingController()..addListener(_onSearchPokemons);
+    isSearching = false;
+  }
+
+  @override
+  void dispose() {
+    _onClearSearchedPokemons();
+    _searchController.dispose();
+    _debounceTimer?.cancel();
+
+    super.dispose();
+  }
 
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(title: const Text(appName)),
-      body: Center(
-        child: pokemons.when(
-          (data) => GridView.builder(
-            itemCount: data.length,
-            gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
-              crossAxisCount: gridRowCount,
-              childAspectRatio: cardRatio,
+      body: Container(
+        padding: const EdgeInsets.all(10.0),
+        child: Column(
+          children: [
+            SearchBar(
+              searchController: _searchController,
+              onClearSearchText: _clearSearchedPokemons,
+            ),
+            const VerticalSpace(height: 20.0),
+            Expanded(
+              child: isSearching
+                  ? PokemonGridView(
+                      pokemons: widget.searchedPokemons,
+                      shouldShowFallbackLabel: true,
+                    )
+                  : widget.pokemons.when(
+                      (data) => PokemonGridView(pokemons: data),
+                      loading: () => const Center(child: CircularProgressIndicator()),
+                      error: (errorMessage) {
+                        WidgetsBinding.instance.addPostFrameCallback(
+                          (_) => _showErrorMessageSnackbar(context, errorMessage ?? emptyString),
+                        );
+                        return const BodySmallText(text: errorMessageDefault);
+                      },
+                    ),
             ),
-            itemBuilder: (_, int index) {
-              final pokemon = data[index];
-              return PokemonCard(pokemon: pokemon);
-            },
-          ),
-          loading: () => const CircularProgressIndicator(),
-          error: (errorMessage) {
-            WidgetsBinding.instance
-                .addPostFrameCallback((_) => _showErrorMessageSnackbar(context, errorMessage ?? emptyString));
-            return const Text(errorMessageDefault);
-          },
+          ],
         ),
       ),
     );
   }
+
+  void _clearSearchedPokemons() {
+    widget.onClearSearchedPokemons;
+    _searchController.text = emptyString;
+    setState(() => isSearching = false);
+  }
+
+  void _onClearSearchedPokemons() {
+    if (widget.searchedPokemons.isNotEmpty) widget.onClearSearchedPokemons;
+  }
+
+  void _onSearchPokemons() {
+    if (_debounceTimer?.isActive ?? false) _debounceTimer?.cancel();
+    _debounceTimer = Timer(const Duration(milliseconds: 1000), () => _searchPokemons());
```

Comment:
just shorten this like this: `Timer(const Duration(milliseconds: 1000), _searchPokemons);` 😉